src/pytorch_lightning/tuner/batch_size_scaling.py

```diff
@@ -166,7 +175,16 @@ def _run_binsearch_scaling(
     count = 0
     while True:
         garbage_collection_cuda()
-        trainer.fit_loop.global_step = 0  # reset after each try
+
+        # reset after each try
+        if model.automatic_optimization:
+            optim_progress = trainer.fit_loop.epoch_loop.batch_loop.optimizer_loop.optim_progress
+        else:
+            optim_progress = trainer.fit_loop.epoch_loop.batch_loop.manual_loop.optim_step_progress
+
+        optim_progress.reset()
+        trainer.fit_loop.epoch_progress.current.reset()
```

Comment:
```suggestion
        trainer.fit_loop.epoch_progress.reset_on_run()
```

Reply:
shouldn't be reset? we need to make sure it's a fresh start when running a new batch.

Reply:
That's what `reset_on_run` does, plus resets other state that should be reset.

Reply:
`reset_on_run` doesn't reset the total, right? which we should?

Reply:
Ah you changed it, the diff was doing just `current.reset()` before